module/zfs/vdev_raidz.c

```diff
@@ -2436,6 +2588,19 @@ vdev_raidz_io_start_read_row(zio_t *zio, raidz_row_t *rr, boolean_t forceparity)
 			rc->rc_skipped = 1;
 			continue;
 		}
+		/*
+		 * Check if a data colummn read should be skipped
+		 */
+		if (parity_avail > 0 &&
+		    c >= rr->rr_firstdatacol &&
+		    rr->rr_missingdata == 0 &&
+		    vdev_skip_latency_outlier(cvd, zio->io_flags)) {
```

Comment:
Here seems to be `O(n^2)`.  For each vdev you take a lock and compare it to all other vdevs.